### ISSUE TITLE
Add option to turn off writing plan files

### ISSUE THREAD

**Issue by dreamflasher** [Azure/azure-storage-azcopy#2040]
This is a follow up from this issue: https://github.com/Azure/azure-storage-azcopy/issues/221
All details are provided in length in there, but to summarize:
When transferring a large number of files (>millions) the plan files become so huge that they fill GB to TB of disk space, and thus any job with a large number of files is doomed to fail. 

The solution is to provide an option to turn off writing plan files: https://github.com/Azure/azure-storage-azcopy/issues/221#issuecomment-1120156720

**Comment by JohnRusk** [comment]
@dreamflasher, thinking back to my time working on AzCopy, I can think of several significant difficulties in adapting the current codebase to work without plan files - around retry of failures (single failure or unexpected app termination) and showing percentage complete.  Would you be open to solutions that either:
(a) made the plan files much smaller (e.g. compression) or
(b) offloaded storage of (most of) the plan files to a blob storage account of your choice?

(As noted in the issue you linked from, I'm not involved in the coding on AzCopy anymore, but figured I'd ask these questions here in case they helped).

**Comment by dreamflasher** [comment]
Thank you very much @JohnRusk for the clarifying questions!
As a user of azcopy the thing I care most about is that the files are being transferred correctly. Currently, this is not the case – when trying to transfer a large number of files, plan (and log files) take so much space that the jobs don't complete.
So as a user I don't care how this goal is achieved.

As a developer I would ask: 
* Why do these files need to be so huge? 
* Apparently they grow (at least) linear with the number of files. This is a design flaw. Compression won't solve this. Pruning would. So: Can those files be pruned during the transfer to keep the size constant?

Offloading to storage seems a bad choice, as a) terabytes of plan files become costly, b) transferring those plan files would take as long as executing the azcopy job.

**Comment by JohnRusk** [comment]
@dreamflasher 

This would only be correct if your files were exceptionally small (e.g. <= 10s of bytes per file)

> transferring those plan files would take as long as executing the azcopy job.

**Comment by dreamflasher** [comment]
Fair enough, equal time was an exaggeration, but my hunch is still that it would take quite some time to transfer the plan files and thus significantly slowing down the overall transfer speed.

**Comment by marc-hb** [comment]
> several significant difficulties in adapting the current codebase to work without plan files - around retry of failures

OK but what about successful transfers?  I bet very few people want the ability to retry successful transfers.

We have a wrapper script that runs `azcopy jobs clean --with-status Completed` after using `azcopy`. Why do most users have to implement the same wrapper script?

**Comment by JustinGrote** [comment]
I ran into a version of this when trying to delete a TON of files (100 million). Had to make my own program, hope it helps someone: https://github.com/justingrote/azblobdeletefast
